test/common/common/utility_test.cc

```diff
@@ -875,4 +875,29 @@ TEST(InlineStorageTest, InlineString) {
   EXPECT_EQ("Hello, world!", hello->toString());
 }
 
+TEST(ContainerRemoveElementsTest, Containers) {
+  auto l = StringUtil::splitToken("one,two,three", ",");
+  std::function<bool(const absl::string_view&)> onep = [](const absl::string_view& s) {
```

Comment:
you can use `auto` for the type of `onep` here & below

Reply:
I thought so too, but I immediately end up in template heck:
`bazel-out/k8-fastbuild/bin/source/common/common/_virtual_includes/utility_lib/common/common/utility.h:726:15: note: candidate template ignored: could not match 'function<bool (const type-parameter-0-1 &)>' against '(lambda at test/common/common/utility_test.cc:880:15)'
  static void removeMatchingElements(Container& c, std::function<bool(const T&)> predicate) {
`

any ideas?

Reply:
Flailing here: usually we pass string_view by value, not const ref What happens if you change to that?

Reply:
A slightly more decipherable error:
`utility.h:726:15: note: candidate template ignored: could not match 'function<bool (const type-parameter-0-1 &)>' against '(lambda at test/common/common/utility_test.cc:880:15)'`

I presume we want the predicate signature to stay const ref for efficiency.

Reply:
Yes but that can be specified at the call-site. Often with templates we are as general as possible. In this case imagine that you have a `vector<int>`; it would be less efficient to pass those to the predicate by reference. But if you have `vector<unique_ptr<string>>` then I think you *must* use const ref but the caller can specify it.

Reply:
gotcha, thanks. Changing to `std::function<bool(T)>` doesn't make it work with `auto` (or inline lambdas for that matter), but it does make more sense at least.